File: awa/sdk_dist/csharp/Awa.Client/Utilities/Activity/IsDirectoryActivity.cs
using System;
using System.Threading.Tasks;
using Awa.Client.Constants;

namespace Awa.Client.Utilities.Activity
{
    public static partial class ActivityUtilities
    {
        /// <summary>
        /// Check if a path is a directory.
        /// </summary>
        /// <param name="dirPath">The directory path to check.</param>
        /// <returns>True if the path is a directory, False otherwise.</returns>
        public static async Task<bool> IsDirectoryActivity(string dirPath)
        {
            return await Temporalio.Workflows.Workflow.ExecuteActivityAsync<bool>(
                AwaConstants.ActivityIsDirectory,
                new object?[] { dirPath },
                new()
                {
                    TaskQueue = AwaConstants.AwaDefaultTaskQueue,
                    StartToCloseTimeout = TimeSpan.FromSeconds(AwaConstants.FileIoTimeoutSeconds)
                });
        }
    }
}

File: awa/awa/sdk/utils/activity/is_directory_activity.py
from datetime import timedelta
from pathlib import Path

from temporalio import workflow

from awa.sdk import constants


async def is_directory_activity(dir_path: str | Path) -> bool:
    """Check if a path is a directory.

    Args:
        dir_path: The directory path to check.

    Returns:
        True if the path is a directory, False otherwise.

    """
    return await workflow.execute_activity(
        constants.ACTIVITY_IS_DIRECTORY,
        arg=str(dir_path),
        task_queue=constants.AWA_DEFAULT_TASK_QUEUE,
        start_to_close_timeout=timedelta(seconds=constants.FILE_IO_TIMEOUT_SECONDS),
    )
